packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -1774,6 +1774,39 @@ describe('TransactionController', () => {
 
       expect(delayMessengerMock.call).not.toHaveBeenCalled();
     });
+
+    it.each([
+      [
+        providerErrors.userRejectedRequest(),
+        'catches error when user reject approval',
+      ],
+      [
+        new Error('TestError'),
+        'catches error no code property in error object',
+      ],
```

Comment:
Additional coverage is great, but should we also add an `undefined` object as an error to cover the bug scenario?

Reply:
Done. :pray: